File: src-model/model/DrawableResolution.java
package model;

public enum DrawableResolution {

    VERY_CLOSE(64, 25),
    CLOSE(32, 50),
    MEDIUM(24, 100),
    FAR(12, 400),
    VERY_FAR(8, 1200),
    INVISIBLE(0, 0);

    /* ========== PRIVATE ========== */
    private double distance;
    private int resolution;
    private int ringResolution;

    private DrawableResolution(int resolution, double distance) {
        this.resolution = resolution;
        this.ringResolution = 4 * resolution;
        this.distance = distance;
    }

    public int getResolution() {
        return resolution;
    }

    public int getRingResolution() {
        return ringResolution;
    }

    public static DrawableResolution determineResolution(double radius, double distanceBetween) {

        double ratio = distanceBetween / radius;

        // Let's assume that enums are sorted
        for (DrawableResolution resolution : values()) {
            if (ratio < resolution.distance) {
                return resolution;
            }
        }

        return INVISIBLE;
    }
}

File: src-model/model/type/DrawableResolution.kt
package model.type

enum class DrawableResolution(
        private val resolution: Int,
        private val distance: Float
) {

    VERY_CLOSE(64, 1f),
    CLOSE(32, 0.1f),
    MEDIUM(24, 0.01f),
    FAR(12, 0.001f),
    VERY_FAR(8, 0.0001f),
    INVISIBLE(0, 0.0f);

    fun distance(): Float {
        return distance
    }

    fun resolution() : Int {
        return resolution
    }

    companion object {

        fun determineResolution(ratio : Float) : DrawableResolution {
            return values().firstOrNull { ratio > it.distance() } ?: DrawableResolution.INVISIBLE
        }
    }
}